scripts/zkai_larger_native_boundary_candidate_selector_gate.py

```
@@ -0,0 +1,704 @@
+            raise LargerNativeBoundaryCandidateSelectorError(f"unknown candidate row {candidate_id!r}")
+        for key, expected_value in expected.items():
+            if row.get(key) != expected_value:
+                raise LargerNativeBoundaryCandidateSelectorError(
+                    f"{candidate_id} {key} drift"
+                )
+    if tuple(ordered_ids) != tuple(EXPECTED_CANDIDATES):
+        raise LargerNativeBoundaryCandidateSelectorError("candidate order drift")
+
+    artifacts = payload.get("source_artifacts")
+    if not isinstance(artifacts, list) or len(artifacts) != len(EXPECTED_CANDIDATES) + 1:
+        raise LargerNativeBoundaryCandidateSelectorError("source artifact count drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)